ocs_ci/utility/tnf_hypervisor.py

```
@@ -0,0 +1,937 @@
+                logger.info(f"{resource_name} deleted")
+                return True
+            except Exception as e:
+                if "DependencyViolation" in str(e) and attempt < retries - 1:
+                    logger.info(
+                        f"{resource_name} has dependencies, "
+                        f"retrying in {delay}s ({attempt + 1}/{retries})..."
+                    )
+                    time.sleep(delay)
+                else:
+                    logger.warning(f"Failed to delete {resource_name}: {e}")
+                    return False
```

Ruff (0.16.1)

[warning] Missing explicit `return` at the end of function able to return non-`None` value

Add explicit `return` statement

(RET503)